lib/screens/home/screen.dart

```diff
@@ -19,25 +19,37 @@ class _HomeScreenState extends State<HomeScreen> {
     super.initState();
 
     // Fetch currently valid T&C version unconditionally when initializing the widget.
-    final tacAcceptance = context.read<TermsAndConditionAcceptance>();
-    final network = context.read<ActiveNetwork>();
-    final services = context.read<ServiceRepository>().networkServices[network.active]!;
-    _updateValidTac(services.walletProxy, tacAcceptance);
+    // TODO: Use the 'tacAcceptance.state.valid.updatedAt' to determine whether to perform the refresh
+    //       (now and on other appropriate triggers like activating the app).
+    _refresh(context, context.read<TermsAndConditionAcceptance>());
   }
 
   static Future<void> _updateValidTac(WalletProxyService walletProxy, TermsAndConditionAcceptance tacAcceptance) async {
     final tac = await walletProxy.fetchTermsAndConditions();
     tacAcceptance.validVersionUpdated(ValidTermsAndConditions.refreshedNow(termsAndConditions: tac));
   }
 
+  void _refresh(BuildContext context, TermsAndConditionAcceptance tacAcceptance) {
+    final network = context.read<ActiveNetwork>().state;
+    final services = context.read<ServiceRepository>().networkServices[network.active]!;
+    _updateValidTac(services.walletProxy, tacAcceptance);
+  }
+
   @override
   Widget build(BuildContext context) {
+    final tacAcceptance = context.watch<TermsAndConditionAcceptance>();
     return Scaffold(
       body: Container(
         padding: const EdgeInsets.fromLTRB(16, 64, 16, 16),
-        child: Builder(
-          builder: (context) {
-            final tacState = context.watch<TermsAndConditionAcceptance>();
+        child: BlocConsumer<TermsAndConditionAcceptance, TermsAndConditionsAcceptanceState>(
+          bloc: tacAcceptance,
```

Comment:
```suggestion
```

We can omit the `bloc` since `BlocConsumer` will do the lookup.

Reply:
done